backend/src/main/java/ba/unsa/etf/si/bbqms/ws/controllers/TellerStationController.java

```diff
@@ -153,6 +154,20 @@ public ResponseEntity getBranchStations(@PathVariable final String tenantCode,
         }
     }
 
+    @GetMapping("/{stationId}/tickets")
+    public ResponseEntity getTicketsForTellerStation(@PathVariable long stationId) {
+        try {
+            Set<Service> services = stationService.getServicesForTellerStation(stationId);
+            Set<Ticket> tickets = ticketService.getTicketsForServices(services);
+            List<TicketDto> ticketDtos = tickets.stream()
+                    .map(TicketDto::fromEntity)
+                    .collect(Collectors.toList());
+            return ResponseEntity.ok().body(ticketDtos);
+        } catch (Exception e) {
+            return ResponseEntity.badRequest().build();
+        }
+    }
+
```

Comment:
We **must not** fetch all tickets which have a certain service like this, because in that case we will fetch some tickets which require that service, but in another branch, which is wrong.

When fetching the tickets we must choose only those which belong to this station's branch.

In addition to this, we should not fetch tickets which are already assigned to a different teller station.

Please test all of these cases thoroughly.